server/pulp/server/db/model/__init__.py

```diff
@@ -597,3 +607,147 @@ class CeleryBeatLock(Document):
+    :ivar password: encrypted password for login credentials
+    :type password: basestring
+    :ivar roles: list of roles user belongs to
+    :type roles: list of str
+    :ivar _ns: (Deprecated), Contains the name of the collection this model represents
+    :type _ns: mongoengine.StringField
+    """
+
+    login = StringField(required=True, regex=r'^[.\-_A-Za-z0-9]+$')
+    name = StringField()
+    password = StringField()
+    roles = ListField(StringField())
+
+    # For backward compatibility
+    _ns = StringField(default='users')
+
+    meta = {'collection': 'users',
+            'allow_inheritance': False,
+            'indexes': ['-roles', {'fields': ['-login', '-name'], 'unique': True}],
+            'queryset_class': CriteriaQuerySet}
+
+    serializer = serializers.User
+
+    def is_superuser(self):
+        """
+        Return True if the user with given login is a super user
+
+        :return: True if the user is a super user, False otherwise
+        :rtype:  bool
+        """
+        return SUPER_USER_ROLE in self.roles
+
+    def set_password(self, plain_password):
+        """
+        Sets the user's password to a hashed version of the plain_password. This does not save the
+        object.
+
+        :param plain_password: plain password, not hashed.
+        :type  plain_password: str
+
+        :raises pulp_exceptions.InvalidValue: if password is not a string
+        """
+        if plain_password is not None and not isinstance(plain_password, basestring):
+            raise exceptions.InvalidValue('password')
+        self.password = self._hash_password(plain_password)
+
+    def check_password(self, plain_password):
+        """
+        Checks a plaintext password against the hashed password stored on the User object.
+
+        :param plain_password: plaintext password to check against the stored hashed password
+        :type  plain_password: str
+
+        :return: True if password is correct, False otherwise
+        :rtype:  bool
+        """
+        salt, hashed_password = self.password.split(",")
+        salt = salt.decode("base64")
+        hashed_password = hashed_password.decode("base64")
+        pbkdbf = self._pbkdf_sha256(plain_password, salt, PASSWORD_ITERATIONS)
+        return hashed_password == pbkdbf
+
+    def _hash_password(self, plain_password):
+        """
+        Creates a hashed password from a plaintext password.
+
+        _hash_password, check_password, _random_bytes, and _pbkdf_sha256 were taken from this
+        stackoverflow.com : http://tinyurl.com/2f6gx7s
+
+        :param plain_password: plaintext password to be hashed
+        :type  plain_password: str
+
+        :return: salt concatenated with the hashed password
+        :rtype:  str
+        """
+        salt = self._random_bytes(8)  # 64 bits
+        hashed_password = self._pbkdf_sha256(str(plain_password), salt, PASSWORD_ITERATIONS)
+        # return the salt and hashed password, encoded in base64 and split with ","
```

Comment:
Be careful adding comments that simply state in English what the code is doing.  It just adds code-noise.
Eg:
```
# add two numbers and assign to a variable
a = b + c
```
As the code reader, I have to stop processing code in my mind, context switch to processing English, then back to processing code.  I *can* make the code more effort to read.

I suspect you've never heard this and surprised I am suggesting you actually remove a comment.  Just suggesting that more comments are not always better.  There is a cost to the reader.